src/components/shared/chart/Canvas.js

```diff
@@ -30,6 +30,9 @@ type Props<HoveredItem> = {|
   // Default to true. Set to false if the chart should be redrawn right away after
   // rerender.
   +drawCanvasAfterRaf?: boolean,
+
+  +onMouseMove?: MouseEventHandler,
+  +onMouseLeave?: MouseEventHandler,
```

Comment:
Currently, your design makes these functions take a DOM `MouseEvent` (and you pass it `nativeEvent`). I think it would be less surprising if we passed the React event instead. To access `offsetX` I believe we still need to use `nativeEvent` (at least according to Flow) so that shouldn't change your code otherwise.